library/std/src/net/hostname.rs

```diff
@@ -0,0 +1,34 @@
+use libc::gethostname;
+use std::ffi::CStr;
+use std::ffi::OsString;
+use std::os::raw::c_char;
+
+/// Returns the system hostname.
+///
+/// The returned result will, on success, return the same result [`libc::gethostname`] would return
+/// (as it is implemented using the very same function), and on error, what `errno` contains, also
+/// set by [`libc::gethostname`].
+#[stable(feature = "rust1", since = "1.84.0")]
```

Comment:
this needs to be gated behind a new unstable feature gate associated with a tracking issue, and `since` should be set to `CURRENT_RUSTC_VERSION` since this is not going to be stabilized in 1.84 which is stable in 4 days.

Reply:
oop, I figured it was okay to do this. I guess not.